vnext/Microsoft.ReactNative/ReactHost/CrashManager.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "CrashManager.h"
+#include "ReactInstanceWin.h"
+
+#include <winrt/Windows.Storage.h>
+
+#include <WerApi.h>
+
+namespace Mso::React {
+
+// when calling SetUnhandledExceptionFilter the previous filter is returned
+// this should be saved so that if needed it can be called after the new unhandled exception has finished executing
+// this allows the Windows Error Reporting system to process the error and upload the data for processing
+LPTOP_LEVEL_EXCEPTION_FILTER g_previousExceptionFilter = nullptr;
+
+std::wstring g_logFileName;
+bool g_wasCalled{false}; // Prevent recursion in the custom handler
+
+extern "C" LONG WINAPI CustomWERExceptionFilter(LPEXCEPTION_POINTERS const exceptionPointers) {
+  if (g_wasCalled) {
+    return EXCEPTION_EXECUTE_HANDLER;
```

Comment:
>EXCEPTION_EXECUTE_HANDLER [](http://example.com/codeflow?start=11&length=25)

This looks like a we are in really bad state if there is recursion. Could you add comment on the expected behavior if that happens?

Reply:
an exception handler can, in turn, cause an exception :)
I think in some cases, double faults should just kill the app, it's unlikely we'll recover gracefully enough

Reply:
Based on the documentation ([https://docs.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-unhandledexceptionfilter](https://docs.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-unhandledexceptionfilter)) it looks like this is the only value we can return here to avoid recursing. They suggest returning to allow the exception handler to do its thing; the alternative is to terminate the process, yes, though not sure if that'll allow WER to finish its collecting.